tests/integration/suite/test_cluster_scan.py

```diff
@@ -0,0 +1,18 @@
+from .common import random_str
+from .conftest import cluster_and_client
+
+def test_run_scan_not_available_on_not_ready_cluster(admin_mc):
+    client = admin_mc.client
+    cluster = client.create_cluster(
+        name=random_str(),
+        rancherKubernetesEngineConfig={
+            "accessKey": "junk"
+        }
+    )
+    _, cluster_client = cluster_and_client(cluster.id, client)
+    try:
+        admin_mc.client.action(obj=cluster, action_name="runSecurityScan", )
+    except AttributeError as e:
+        assert e is not None
+    client.delete(cluster)
```

Comment:
To ensure cluster object is cleaned up no matter of runtime errors, we should call remove_resource() like here https://github.com/rancher/rancher/blob/master/tests/integration/suite/test_clustertemplate.py#L688

Reply:
also the test seems to have flake8 errors

Reply:
@prachidamle fixed.

Reply:
`client.delete(cluster)` is redundant with the `remove_resource` call above so it can be dropped.

Reply:
Will fix.